packages/driver/src/cypress/browser.js

```diff
@@ -2,21 +2,60 @@ const _ = require('lodash')
 const $utils = require('./utils')
 const $errUtils = require('./error_utils')
 
-const isBrowser = function (config, obj = '') {
-  if (_.isString(obj)) {
-    const name = obj.toLowerCase()
-    const currentName = config.browser.name.toLowerCase()
+const _isBrowser = function (browser, matcher, errPrefix) {
+  let isMatch
+  let exclusive = false
 
-    return name === currentName
+  const matchWithExclusion = (objValue, srcValue) => {
+    if (srcValue.startsWith('!')) {
+      exclusive = true
+
+      return objValue !== srcValue.slice(1)
+    }
+
+    return objValue === srcValue
+  }
+
+  if (_.isString(matcher)) {
+    const name = matcher.toLowerCase()
+    const currentName = browser.name.toLowerCase()
+
+    isMatch = matchWithExclusion(currentName, name)
+  } else if (_.isObject(matcher)) {
+    isMatch = _.isMatchWith(browser, matcher, matchWithExclusion)
+  } else {
+    $errUtils.throwErrByPath('browser.invalid_arg', {
+      args: { prefix: errPrefix, obj: $utils.stringify(matcher) },
+    })
   }
 
-  if (_.isObject(obj)) {
-    return _.isMatch(config.browser, obj)
+  return {
+    isMatch,
+    exclusive,
   }
+}
+
+const isBrowser = function (config, obj = '', errPrefix = '`Cypress.isBrowser()`') {
+  if (!_.isArray(obj)) obj = [obj]
+
+  const result = obj.map((matcher) => _isBrowser(config.browser, matcher, errPrefix))
+  .reduce((a, b) => {
```

Comment:
Can we use more descriptive variable names here than `a` and `b`? What is a and b?

Reply:
I prefer to always use a,b for reduce functions. feel free to suggest different names